### ISSUE TITLE
PART category

### ISSUE THREAD

**Issue by nschneid** [UniversalDependencies/UD_Hebrew#10]
[`PART`](http://universaldependencies.org/he/pos/PART.html) is used for את, which makes sense.
But it is also frequently used for several things that strike me as odd:

- אותו, etc.: Why not `PRON`? Or else split into `PART` + `PRON`?
- של: Why not `ADP`?
- שלו, etc.: Why not `PRON`? Or else split into `ADP` + `PRON`?
- כש: Why not `SCONJ`?

There's also the rare word מש—not sure what should be done with that.

**Comment by yoavg** [comment]
If I remember correctly (and perhaps @rtsarfaty can confirm / remember differently), the rationale behind של and כש is that they assign case.

Re אותו, note that the complete analysis is (taking a random example from the training set):
```
33-34   אותו    _       _       _       _       _       _       _       _
33      אותו    _       PART    PART    Case=Acc        34      case:acc        _       _
34      הוא_    _       PRON    PRON    Gender=Masc|Number=Sing|Person=3|PronType=Prs   32      dobj    _       _
```

So we treat it as a two-words token (את הוא), where the first is indeed PART and the second is PRON (which is what you suggest).

The debate could be regarding representing the first word as אותו and the second as הוא_ rather than את followed by הוא. There are arguments to each directions, but users of the treebank often find this representation more natural and convenient, as it it easier to align back to the sentence text.

**Comment by nschneid** [comment]
OK, I understand אותו then.

Regarding של: what is the evidence that it assigns case differently from other prepositions, like -ל and על? Or English "of"?

Regarding כש: I thought it normally took a clausal complement, like English "when".

**Comment by rtsarfaty** [comment]
Thanks Nathan for your question, I think it conflates several different issues, or conventions, let me comment on them separately:

1. Concerning the two-level representation:

1a. words such as אותו, שלנו, בשבילך, לי are segmented in the treebank to two separate words/lexical items in this fashions:
אותו --> את + הוא
שלנו --> של + אנחנו
בשבילכם --> בשביל + אתם
לי --> ל + אני
the reasons and arguments for that separation are detailed in the original hebrew tree banking paper
http://www.phil.uu.nl/~yoad/papers/SimaanEtAl.pdf
and the key point is to allow tree structures that are not DAGs, because the two items will attach at different level. you can see further discussion of the non-segmented alternative of these items here
http://tsarfaty.com/pdfs/lrec08.pdf

1b. In retrospect, not much had changed since the original annotation of the first hebrew treebank, even in view of the alternatives. The only change that was advanced is representing the host את as אותו to help aligning back to the original text (because there is no deterministic process / FSM to combine את+הוא into אותו -- most of these fusions, for different prepositions, are idiosyncratic). 
So what you currently see is:
אותו --> אותו + הוא
שלנו --> שלנו + אנחנו
בשבילכם --> בשבילכם + אתם
לי --> לי + אני
(I agree it has drawbacks -- in fact I prefer the 1a representation, though it is not a strong preference -- happy to hear what others think),
But the essential logic behind the segmentation and two-level representation, as in the original treebank, is the same.

2. Tagging conventions:

2a. In all of the above cases you can view the host (first item) as a preposition that assigns case to the pronoun:
שלנו --> של + אנחנו
של=of, 
אנחנו=us, feats: first person plural

בשבילכם --> בשביל + אתם
בשביל=for 
אתם=you, feats: second person plural

לי --> ל + אני
ל=to 
אני=me,  feats: first person singular

The prepositions here may be seen as essentially adding case to the pronoun, indicating its function in the syntactic representation: "of" may be seen as assigning genitive case, "for" indicates a purpose,  "to" indicates a direction/locative, etc. Note that these prepositions can accept any nominal host, not only pronouns (including PROPN, construct state nouns and MWEs) and they would still be functioning the same and represented the same. But some of them are standalone (בשביל) and some of them are prefixes that must attach to the host (ל)

As case-assigning prefixes, one could advance arguments for tagging them as PART that carry case, or they can be tagged as adpositions in parallel to other case assigning prepositions in other. In view of Manning's law, I think that the latter option, tagging them (and all other prepositions, prefixed and standalone) as ADP is more justified, as (i) this is similar to what happens with their counterparts in well known european languages, leading to (ii) more user- and annotator-friendly, and (iii) probably also more suitable for parsing because it captures a generalization concerning the similar distribution of these items.

2b. Note that אותו is rather different than the rest of the examples, since it has a different (and very restricted) distribution (cf Aissen http://www.jstor.org/stable/4048040?seq=1#page_scan_tab_contents). linguists generally agree that את is purely a case marker, with no parallel preposition in other languages (other than case marking morphemes). Because of the restricted distribution and particular behaviour of את I believe it is justified to leave it as PART with the acc case feature.
 
3. "Clause-level clitics"

Prefixes such as כש מש לכש and so on indicate time subordination roughly corresponding to 
"When <clause>". Again, one could advance arguments to viewing them as temporal case assigners, especially due to the fact that these are prefixed morphemes that attach to a word, and they never stand on their own.  But I think it would be more justified, universally coherent, and more straight forward, (again in accordance to Manning's Law), to tag them as SCONJ (like their counterparts in other languages, and regardless of the fact that in other languages they are standalone lexical items.)

To summarise, the answers to your original questions are then:

1. אותו should be PART+PRON (and in most cases in the treebank it is indeed the case -- if not, it should be fixed)
2. שלו is currently PART+PRON and should be ADP+PRON
3. כש מש לכש etc are currently PART and should be SCONJ

Assuming everyone on the team agrees (@yoavg) and no one outside of the team objects we will advance these changes in v2.

**Comment by yoavg** [comment]
I agree to these changes, and am OK with considering the current situation as a conversion bug -- as such I'll try to incorporate a fix and push in the coming days, already in V1 (unless someone disagrees?)

**Comment by jnivre** [comment]
Fixing and pushing (to the dev branch) in the coming days is fine, but it won't be v1 because the next release will be v2.

**Comment by amir-zeldes** [comment]
Allow me to add my support for @rtsarfaty 's preference for 1a above: the reasoning that

אותו --> אותו + הוא

is more reconstructible seems to me to lead to unwanted variation in the split token level. If your text has variants like אותכם and אתכם, you will end up with two different forms for the split representation, even though we're not dealing with the surface form anymore anyway. Is it really desirable for these cases to be distinct at the sub-token level? Subtokens like this seem more consistent to me:

אותכם --> את + אתם
אתכם --> את + אתם

Similar issues apply to מכם : ממכם, אתנו : עמנו and more. I think in the UD context they are variant results of fusing the same constituent units (a linguistic, morphological account might be substantially more complex).

**Comment by nschneid** [comment]
I agree with the stylistic preference for 1a, because otherwise it looks like there are duplicate words when reading the sentence. As long as the full wordform is present on one of the lines, I don't see why it should be programmatically harder to align to the original sentence. I would expect cross-linguistic consistency to favor 1a as well, judging by the *vámonos* example in the [CoNLL-U format documentation](http://universaldependencies.org/format.html).

**Comment by yoavg** [comment]
Note: the full form is available on lines that are currently NOT predicted by automatic tools.
We used to have a representation similar to the one proposed by @amir-zeldes , and it was a disaster for users who did not know Hebrew and tried to use the corpus, or automatic parse trees.
As a compromise, how about using the representation proposed by @amir-zeldes at the lemma level?

**Comment by nschneid** [comment]
> it was a disaster for users who did not know Hebrew and tried to use the corpus, or automatic parse trees

Is this something that could be fixed by documentation?

I'm not sure the current practice reflects the intention of multiword items in CoNLL-U, or matches their use in other languages. If the goal is for UD to be a new and better standard, it seems odd to adhere to language-specific constraints imposed by prior representations/tools.

If it needs to be made absolutely clear that only the multiword line (e.g. the one numbered "33-34") contains the full wordform, then another solution would be to use underscores on both components:

אותו --> את_ + _הוא

**Comment by amir-zeldes** [comment]
@yoavg I understand why this might have been confusing for some, maybe lemmas could help here. Although I'm not sure a catastrophic failure is a bad idea, maybe it's better if users then notice their mistake, rather than having something that looks almost OK and have people not realize that a sequence like אותו הוא doesn't actually occur (false but plausible looking training data). The underscores suggested by @nschneid could solve this.

ML applied to the subtokens only, without the full forms, may also suffer from distinguishing ממכם and מכם etc. due to data sparseness (unless you're trying to predict register etc.). I think a parser should do better if they are collapsed into the same sub-tokens, since syntactically they are the same.

**Comment by rtsarfaty** [comment]
Thanks @nschneid @amir-zeldes for the feedback. I agree with all you say, but I also know the pains of end users not knowing where the את arrived from when surface shows אותו -- as @yoavg brought up.

Perhaps I may try and suggest a compromise solution -- 
why don't we take the form representation of אותו as @nschneid suggests:
את_ _הוא 
representation, which I like, and extend to something more explicit such as 
את_@אותו  _הוא
The את_@אותו  is to be interpreted as:
 את in the context of אותו 
as the form that is currently being analysed in terms of tags, features, dependencies.

Pros:
- captures the fact that את_@אותה את_@אותנו את_@אותו belong to the paradigm of את  
- captures the fact that the analysed token אותו is the source of the predicted את during analysis
- captures the fact that the predicted הוא_ is different than a simple pronoun הוא in its syntactic (distributional) behaviour (e.g., הוא_ does not allow high non-local attachments while הוא does)
- easy grepable, and the "in the context of ..." can be gotten rid of as needed.

@yoavg @nschneid @amir-zeldes what do you think about that for the hebrew data?

The only "con" I can think of is that we exploit the FORM column format of UD for encoding something slightly more sophisticated and less standard of the two level representation. However, I don't think this is a "bug", this is more like a "feature", to be discussed in the team -- any language with high fusion needs to predict the syntactic words while also being able to point out the token it is originated from, to make the output usable for non speakers of the language that align back to the original text (this need is less pertinent for linguists who seek generalisations, but more for end users interested in applications). I believe we will need some solution for that. This representation may also be seen as a proposed standard for high fusion language wherein the origins of fused forms are unpredictable (and go beyond simple segmentation -- eg. deletion, reduplication, and so on). Further down the road this will also help solve problems in evaluation that have to do with alignment of predicted segmentations with gold segmentation when they differ.

Finally, this issue may be related to various other aspects of segmentation from a cross-linguistic point of view -- in which case, perhaps it should also be discussed in the working group that @jnivre suggested.

**Comment by yoavg** [comment]
Thanks,

Regarding the question "will this be solved by documentation" -- not really, in my experience. There are numerous cases, and most users, definitely casual users, don't read, as much as we would like them to.

Regarding "the parser could work better if..." well, the parser's designer is free to do whatever they want with the tokens before they are fed to the parser. This is in line with the UD principles that "parsability" should not be a concern (arguably the parsers for English will be better if prepositions head PPs).

Regarding the proposals to add underscores or the more elaborate representation by Reut,
I'm ok with both, with a slight preference for the more minimalistic underscores solution, but will not argue either way, really both are fine with me. Note that we currently already have an underscore on the pronoun, which is the artificial element that was added to the sentence.

**Comment by nschneid** [comment]
> a slight preference for the more minimalistic underscores solution

Same here, unless there's an agreement across languages that a more complex FORM column is warranted in situations like this.

I don't think people should be confused about where the _את came from once they look at the sentence and see the multiword indices. The @ solution might add confusion, especially as people not familiar with RTL scripts might not be sure of the order of the forms on either side of it.

**Comment by amir-zeldes** [comment]
I think I also prefer what @nschneid is saying. And I don't think it's that strange, actually very many languages now have similar situations in conllu, for example handling of German or French fused preposition+article:

https://github.com/UniversalDependencies/UD_German/blob/master/de-ud-test.conllu#L59-L61
https://github.com/UniversalDependencies/UD_French/blob/master/fr-ud-test.conllu#L82-L84

A French sequence like `à le` is also ruled out as normal running French and would also mess up a real-world parser, but there's no warning sign whatsoever. Either you look at the supertoken `au` or you need to learn some French!

**Comment by rtsarfaty** [comment]
Okay, I think it's settled then -- we will proceed with the @nschneid proposal. 
This change will be advanced as part of the v2 version of the treebank.

**Comment by yoavg** [comment]
oh, damn, I wasn't paying attention and thought @nschneid 's proposal was:
```
אותו_  _הוא.
```
So I was supporting something I wasn't really meaning to support. But if this is the general consensus, then fine, I will not fight for it.

**Comment by jnivre** [comment]
I don't want to rock the boat now that everyone seems to be converging, but I am not too happy about using underscores as "warning flags". For example, neither French nor German, referred to above, use any special character to indicate that a word has been extracted from a multiword token. Thus, "du" only decomposes into "de" and "le", not "de_" and "_le" or anything like that. One argument for this is that "de" and "le" is grammatically the same as the "de" and "le" that elsewhere occur as multiword tokens and perhaps the situation in Hebrew is different in that these words never occur on their own. Or perhaps I have misunderstood the whole thing. :)

**Comment by rtsarfaty** [comment]
@jnivre as far as I understand you did not misunderstand :)

_הוא and הוא have different behaviour and different distributions. To take a simple example, while הוא can attach, e.g., in subject position, _הוא must be strictly a prepositional object.  Unifying them to a single token will create a de facto generalisation over the data that is not actually correct, and will lead to spurious, and completely redundant, attachment ambiguities, that shouldn't have been there in the first place.

I hope that this clarifies the situation a bit.

**Comment by amir-zeldes** [comment]
Not necessarily drawing any consequences from this, but I think the situation is the same in French: 

The subtoken `le` which becomes something like `_u` in `du` also can't stand except after a preposition. So by that argument, French also has multiple items `le`, and, arguably, a parser might benefit from getting the information that a specific `le` is the one that has to be a determiner of a propositional object. Or am I missing some difference between these languages?

**Comment by rtsarfaty** [comment]
@amir-zeldes 

I am no expert in French and thus will not comment on the actual observation, but I wouldn't be surprised if this is not a difference between the languages but a difference in the structure we (theoretical or computational linguists) impose on the language.

More specifically, theories of word-based morphology (by linguists as Anderson, Stump, Blevins, etc) provide a good explanation / justification as to why isolating the analysis of the segmented items (be it a pronominal clitic or something else) from the token they originated from actually harms the linguistic description (in terms of adequacy, generalisation, economy, etc). According to Stump's taxonomy of morphological models there are multiple views of morphology that one can adopt, but the UD guidelines deliberately select a strong word-based view, and for this reason I think these arguments should be taken seriously.

just my 2p..

**Comment by yoavg** [comment]
@jnivre , @amir-zeldes , to add to @rtsarfaty 's argument, note that in the Hebrew TB, this pronominal element is not restricted to these inflected prepositions/case-markers -- we also have the same analysis for nouns with a genitive suffix 
(עורפו, his-neck, is currently analyzed as:
עורפו + _הוא)

again, the decision was to keep the noun in its inflected form, and also add the suffix as a distinct element.

The reason we need the distinct pronoun, is that (admittedly in some rare cases) it can attach differently from main noun. However, it is not the same as the pronoun הוא, and should receive a different form in my opinion. I would actually rather see it as some completely artificial element (i.e. `*PRN*`) that can take an attachment, but in favor of the word-based approach of the UD, we decided to assign it a real word-like form.
But I maintain that these cases are inflections, not concatenated suffixes (definitely in the prepositions אותו, בשבילו, עמו...).

**Comment by nschneid** [comment]
If I'm not mistaken, the debate has turned to which inflected forms should be decomposed *at all* for syntactic analysis. This is a good discussion to have, of course, but I don't see how it is relevant to the question of how the analysis should be formatted according to the CoNLL-U standard. Where the Hebrew UD treebank chooses to avail itself of the multiword units allowed by CoNLL-U, I think it would make sense to follow the practice in other languages of representing the forms of individual pieces separately rather than giving the inflected form for one of them.

I don't think writing הוא as the form of the suffix portion of אותו or עורפו necessarily constitutes a claim that this is linguistically "the same" as the fully spelled out word הוא appearing as an independent word. (Maybe parsers should have a feature that checks whether the syntactic token is part of a multiword.)

**Comment by amir-zeldes** [comment]
I'm really open with respect to the theoretical analysis and the notation, all that matters to me is being able to find what I'm looking for and convert automatically. I think all of the suggestions above allow that. The only other concern or difference I see between most of these options is cross-linguistic consistency. 

FWIW, Arabic seems to be splitting these, writing them as is (no modification to the substring), and lemmatizing like the independent form. So something like ʕan-hā (about her) is:

```
16-17	عنها	_	_	_	_	_	_	_	_
16	عن	عَن	ADP	P---------	AdpType=Prep	17	case	_	Vform=عَن|Gloss=about,from|Root=`an|Translit=ʿan|LTranslit=ʿan
17	ها	هُوَ	PRON	SP---3FS2-	Case=Gen|Gender=Fem|Number=Sing|Person=3|PronType=Prs	14	nmod	_	Vform=هَا|Gloss=he,she,it|Translit=hā|LTranslit=huwa
```

We're doing the same for Coptic (though the supertokens are actually not in the UD data online yet), and we have a language specific POS distinguishing the bound pronouns from the independent ones (lemma is the same).

**Comment by amir-zeldes** [comment]
Note: in my browser there's an RTL thing going on in the Arabic example above, where the lemma is now displaying to the left of the form - the one showing diacritics is the lemma.

**Comment by yoavg** [comment]
@nschneid it relates to the discussion because, in my view at least, the formatting reflects the analysis.

These forms should be decomposed on syntactic grounds, but I argue that the composition is "full-form + additional-element" and not "hypothesized-reduced-form + hypothesized-suffix" that happen to be words in the language that loosely correlate with the analysis. And I think all inflectional suffixes should receive a similar treatment in the treebank.

I guess this is something to discuss in the "what is a word" working group, and we'll follow whatever decision, but until this is settled, I don't see why Hebrew should follow Arabic on this and not the other way around.

**Comment by nschneid** [comment]
To make sure @jnivre understands the issue, I'll summarize with romanization: 

Prepositions like *šel* 'of' (standalone form) are inflected when taking a pronominal object, e.g., *šelo* 'of.him'. For some prepositions and the ACC case marker for definite NPs, the inflection involves a vowel change in the stem. (For some, if I'm not mistaken, there is suppletion: *bli* 'without' (standalone) vs. *bilʕadi* 'without.me'.)

Using the multiword facility of CoNLL-U, the question is whether this should be represented as šelo → šel + hu (*hu* being the nominative form of the pronoun), or as šelo → šel + o (what Arabic does), or as šelo → šelo + _hu (the current practice), etc.

**Comment by yoavg** [comment]
Thanks for the summary. Note that another possible alternative is `šelo → šelo + *PRN*`.

I will add that it is not always clear what the "base form" should be. For example in "ילדיהן" (their children), should the base form be `ילדי` (can be analyzed as children-of in the smixut construction) or `ילדים` (children) ? The forms `איתם` and `עימם` both mean "with them". For `עימם` the base form is probably `עם` (with), but for `איתם` there is no obvious base form at all (besides maybe using `עם` again, losing the distinction between the two words).  Many other such cases exist.

This prompts me to not decide, and keep the base form as the main form.

In any case the Arabic solution of treating some letters as the pronominal subject does not make much sense to me, and will very likely result in some weird inconsistencies.

**Comment by nschneid** [comment]
I guess yet another option would be šelo → *PREP* + *PRON*. Or essentially, to leave the FORM column blank for the two subparts and only include their lemmas.

The solutions with šelo → šelo + X are counterintuitive to me because it feels like claiming an inflected form consists of itself plus a suffix. If the point is that the inflected forms of *šel* (but not the standalone form) exist in a paradigm, and we don't want to posit an underlying stem for this paradigm and suffix forms for its cells, then it would make sense to have something like:

```
3-4     šelo    šel    ADP     _    2    ...
4       _       hu     PRON    _    3-4  ...
```

where there is no separate line for the stem, but rather, the whole word takes a formless suffix as a syntactic dependent. This would require changes to the CoNLL-U specification.

Now that I think about it, the tokenization of the contracted form *du* → *de* + *le* incorporates a sort of lemmatization step (claiming that *de* and *le* are canonical forms encapsulated in *du*). So there is precedent for include lemmas as the forms within the components of the multiword unit. But I suppose *du* isn't necessarily an instance of paradigmatic inflection like *šeli*, *šelo*, *šela*, etc. Does UD want to develop a principle that contractions like *du* should be treated differently from inflected close-class items like *šelo*?

**Comment by amir-zeldes** [comment]
As @yoavg pointed out, it's not really a closed class problem, since enclitic possessives behave the same way (e.g. beit-o). 

As for the plural token form in possessives, I think it would make sense to use 'yaldei', since it's the same in compounds (smixut): yaldei ha-gan 'children of the kindergarten' also has that form and can have the same meaning (a type of possessive). I think it's just a special morphological form (a 'compounding form'), so it can express being the word 'child' in the lemma, and the word form is inflected (nismax, plural).

Incidentally, Arabic also uses that form for the token in smixut as well as clitic possessives, and there is no supertoken for smixut, as in the nismax in 41 below (mas'uuli- 'officials of...', shortened from mas'uulin):

```
41	مسؤولي	مَسؤُول	NOUN	N------P2R	Case=Gen|Definite=Red|Number=Plur	40	nmod	_	Vform=مَسؤُولِي|Gloss=official,functionary|Root=s_'_l|Translit=masʾūlī|LTranslit=masʾūl
42	الادارة	إِدَارَة	NOUN	N------S2D	Case=Gen|Definite=Def|Number=Sing	41	nmod	_	Vform=اَلإِدَارَةِ|Gloss=administration,management,bureau|Root=d_w_r|Translit=al-ʾidārati|LTranslit=ʾidārat
43	الشيشانية	شِيشَانِيّ	ADJ	A-----FS2D	Case=Gen|Definite=Def|Gender=Fem|Number=Sing	42	amod	_	Vform=اَلشِّيشَانِيَّةِ|Gloss=Chechen|Root=^sI^sAn|Translit=aš-šīšānīyati|LTranslit=šīšānīy
```

I'm not saying Hebrew has to model everything on UD_Arabic and not the other way around, but it would be good to try and do similar things if possible.

**Comment by yoavg** [comment]
@amir-zeldes can you explain this Arabic example for me? Does 41 include a pronominal possessive? if so, where is the morphological properties (Gender,Number) of the pronoun expressed? (if not, then what is the purpose of this example?)

**Comment by nschneid** [comment]
עיר is feminine but its plural is ערים. Does that mean that עריהם would be 'their cities'? (Trying to think of an exception to @amir-zeldes's rule that the smixut form is the same as the plural, minus the ם if it is an ים- plural.)

Hey, is that why it's called a "construct state genitive"? Because it doubles as the stem for pronominal possessive suffixes?

**Comment by yoavg** [comment]
@nschneid 
I don't quite follow the argument, but here is the data:
עריהם is "their(masc) cities". עריהן is "their(fem) cities". ערי is the smixut form of עיר.

**Comment by nschneid** [comment]
OK, then it's not a counterexample to @amir-zeldes's generalization.

**Comment by yoavg** [comment]
I don't understand the generalization and what claim it is trying to support.

**Comment by nschneid** [comment]
@amir-zeldes, in [your earlier comment](#issuecomment-272520073), am I correct in reading that they've given *hua* 'he' as the lemma of the feminine pronominal possessive suffix? I.e. the lemma canonicalizes the gender?

**Comment by yoavg** [comment]
@amir-zeldes just for the heck of it, how would you split רחובותיהן?
רחובות + יהן?
רחובותי + הן?
or something else?
(just to show that the proposal to "write as is", as I understand it, also leads to inconsistencies in the pron forms)

**Comment by nschneid** [comment]
> As for the plural token form in possessives, I think it would make sense to use 'yaldei', since it's the same in compounds (smixut): yaldei ha-gan 'children of the kindergarten' also has that form and can have the same meaning (a type of possessive). I think it's just a special morphological form (a 'compounding form'), so it can express being the word 'child' in the lemma, and the word form is inflected (nismax, plural).

I take this to mean that every noun pluralized with ים- also has the nismax form with י- that also appears with pronominal possessives. So

רחובותיהן --> רחובותי + הן

**Comment by nschneid** [comment]
Oh, I wrote too soon. רחובות is the nismax form. In that case I'm not sure what @amir-zeldes would advocate.

**Comment by yoavg** [comment]
Yup, and compare to תשלומיהן where תשלומי is the nismax form. (edit: I just realized this is redundant with עריהן.)

**Comment by amir-zeldes** [comment]
Hi - the Arabic example is a normal nominal smixut, my only point was that a form like mas'uulii- already has to be used for that, so having it before a possessor pronoun wouldn't be odd. 

The same is true for a majority of Hebrew cases, but there are exceptions. I think for your example I would definitely go with

רחובות
יהן

since that noun form also occurs independently (though I'm really fine with putting underscores or anything else around that). A nastier exception to the generalization @nschneid mentioned is maim : mei- (yam), but also  meimei-xa (and I guess maybe also meimei ha-yam). But that's very unusual in Heb., having this strongly suppletive, totally unpredictable possessed form.

**Comment by rtsarfaty** [comment]
I am not sure why the discussion diverted to noun and in particular to smixut.

Since the first version of the Hebrew treebank (Simaan et al 2001), inflected nouns such as עריהן as well as smixut such as ילדי are not split at all. 
These are simply nouns that carry "two sets" of inflectional features --
ילדי :    NN + [mas+plural] + [smixut]
עריהם : NN + [fem+plural] + [mas + plural]
The first set of features indicates the features of the noun and the second indicates the feature of the possessor(s). This is word-based analysis at play. The discussion of why this is so can be read at the top of page 7 here: http://www.cs.technion.ac.il/~winter/papers/tal.ps. 

A key manifestation of this decision is double genitive constructions in the hebrew treebank, e.g.:
1 תוכניתו 
2 של
3  רז 
which would be analyzed in the treebank:
1 תוכניתו NN+feats1+feats2 
2 של       ADP 
3 רז        PROPN+feats2
And the agreement of the possessor רז will be only with the second layer of features of the noun.
This has been the case in the treebank for all possessed nouns, also in other constructions, for as long as I can remember, and for me this decision is justified.

Now, the case for particles like אותו , preposition like שלנו בשבילו לי, verbs like אהבתיה and so on pose a greater problem because the supposed pronominal clitic can attach at a different location than the host (do have a look at the data, if you want to dig deeper: http://tsarfaty.com/pdfs/lrec08.pdf)
An example is apposition
אותנו הסטודנטים
which is analysed
את
אנחנו
ה
סטודנטים
where [אנחנו הסטודנטים]
is at a lower level of attachment than the particle את

The decision that has been taken (and was never changed) was to split off the clitic, so that the pronoun can attach were it should, separately from the particle. 

At some point when trying to move to complete word-based analysis we considered 
The analysis of שלנו as 
1-2  שלנו
1 שלנו      ADP 
2 *PRO* PRON + 3rd+plural

This makes sense not because *PRO* is an empty suffix, like @nschneid mentioned, but because the pronominal is *missing* and yet is used elsewhere -- just like pro drop languages or missing PRO in relative clauses -- and this is no different than LDDs or "movement" of the pronouns in other contexts/languages. We did not  go for it because most parsers do not recover empty nodes (nor evaluate them) and then we would be missing an important part of the data.

So we have stayed, for the inflected forms of the categories ADP PART VB, with the split analysis that impose implicit lemmatisation:
1-2  שלנו
1 של     ADP 
2 אנחנו PRON + 3rd+plural

This has been confusing to parsers, because of the arguments advanced earlier on different distributions for the standalone ADPs, and so it was later changed to
1-2  שלנו
1 f1(של)     ADP 
2 f2(אנחנו) PRON + 3rd+plural
where f1 and f2 are functions that change the form (using, underscore "_", string context "@suff", etc) 

Different versions of the treebank nowadays have different manifestation of f1 and f2.
It would be great if this could be standardized in UD -- not only for Hebrew but also to all languages that share such challenges

As a side note, and re Arabic, I do think some work needs to be done on making sure analyses are consistent within language families, for instance, Semitic, but it should follow from first-hand UD principles, and not by arbitrarily deciding that one language must follow the other analysis simply bc it currently exists in another treebank.

**Comment by nschneid** [comment]
Interesting—I had not seen this analysis about אותנו הסטודנטים.

If there is not a clearly thought-out policy on how these phenomena are to be handled in UD languages generally, then I think the underscore solution of šelo → šel_ + _hu (this is a manifestation of the f1, f2 solution, right?) would be an improvement over the current policy. Though it would be nice in the long run to come to a cross-lingual consensus. :)

**Comment by rtsarfaty** [comment]
@nschneid 
yup, precisely (re the f1,f2 version). 

I also think that:
-  šelo → šel_ + _hu is a good solution for now for Hebrew 
- would be nice to have a multilingual consensus :)

**Comment by yoavg** [comment]
I agree with all, and am in favor of šelo → šel_ + _hu for Hebrew and in general.

However, re the nouns, the current version of the Hebrew UD treebank does split them, also in the double genitive case. For example:
https://github.com/UniversalDependencies/UD_Hebrew/blob/master/he-ud-dev.conllu#L811

A supporting case for the split analysis would be the same as the in אנחנו הסטודנטים:
```
לדעתנו, המשפחות והחברים, נעשה כאן עוול.
```
Here, המשפחות והחברים is appositive with the pronoun:
```
ל+דעת+(אנחנו, המשפחות והחברים), נעשה כאן עוול.
```

**Comment by amir-zeldes** [comment]
Also +1 for the suggested solution, it looks transparent and the underscores are easy to remove if anyone doesn't like them. I also agree with @yoavg 's apposition analysis.

**Comment by rtsarfaty** [comment]
Thanks all, I would like to recap the decisions with respect to @nschneid's original question:

- אותו 
should be PART+PRON (where it is not the case, will be fixed)

- שלו 
is currently PART+PRON and should be ADP+PRON (will be implemented in v2)

- כש מש לכש 
are currently PART and should be SCONJ (will be implemented in v2)

- אותו, שלו, ביתינו
verbs, prepositions, particles and nouns with possessive clitics will be annotated as two syntactic words, with different features and different attachment preferences, as discussed above, under the following scheme:
1-2 host+clitic
1 host_  (+tag+case features if any)
2 _pron (+tag+pronominal inflectional features: gender, number, person)
(note, when the host is a noun it may carry its one set of pronominal features)

**Comment by jnivre** [comment]
Linguistically, this sounds very good to me. I am still not 100% convinced that the underscores are justified in a cross-linguistic perspective, but since there are no specific guidelines about this in v2, let us accept it for now and make a harmonisation effort for v3. :)
